Explorer/Assets/DCL/CharacterCamera/Systems/UpdateCursorInputSystem.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using DCL.CharacterCamera.Components;
+using DCL.Input;
+using DCL.Input.Systems;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace DCL.CharacterCamera.Systems
+{
+    [UpdateInGroup(typeof(InputGroup))]
+    public partial class UpdateCursorInputSystem : UpdateInputSystem<CameraInput, CameraComponent>
+    {
+        private readonly IEventSystem eventSystem;
+        private readonly DCLInput.CameraActions cameraActions;
+        private readonly PointerEventData pointerData;
+        private readonly List<RaycastResult> results;
+
+        internal UpdateCursorInputSystem(World world, DCLInput dclInput, IEventSystem eventSystem) : base(world)
+        {
+            this.eventSystem = eventSystem;
+            cameraActions = dclInput.Camera;
+
+            // Cache data to avoid allocations
+            pointerData = eventSystem.GetPointerEventData();
+            results = new List<RaycastResult>();
+        }
+
+        protected override void Update(float t)
+        {
+            UpdateInputQuery(World);
+        }
+
+        [Query]
+        private void UpdateInput(ref CameraInput cursorInput)
+        {
+            var mousePos = cameraActions.Point.ReadValue<Vector2>();
+
+            if (cameraActions.Lock.WasPerformedThisFrame() && !cursorInput.IsCursorLocked)
+            {
+                pointerData.position = mousePos;
+                eventSystem.RaycastAll(pointerData, results);
```

Comment:
I have two questions here:

1) Why do you store `pointerData` in this class? Since you have the `UnityEventSystem` class, can't you store it there?
2) This may cause an allocation if the list needs to get resized to put in the results. Is there some way we can estimate a value to set as capacity?

Reply:
Actually the usage of `UnityEventSystem` was the last thing I coded here, it makes sense to move the usage of pointer data over there and probably simplify it even more.
Also pre-allocating the List is a good idea.